File: admin/src/main/java/com/imstargg/admin/domain/Gear.java
package com.imstargg.admin.domain;

import com.imstargg.storage.db.core.MessageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlStarsImageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.GearCollectionEntity;
import jakarta.annotation.Nullable;

import java.util.List;

public record Gear(
        GearCollectionEntity entity,
        List<MessageCollectionEntity> names,
        @Nullable BrawlStarsImageCollectionEntity image
) {
}

File: admin/imstargg-admin-frontend/src/model/Gear.ts
import {MessageEntity} from "@/model/entity/MessageEntity";
import {GearEntity} from "@/model/entity/GearEntity";
import { BrawlStarsImageEntity } from "./entity/BrawlStarsImageEntity";

export interface Gear {
  entity: GearEntity;
  names: MessageEntity[];
  image: BrawlStarsImageEntity | null;
}